api/nodes/views.py

```diff
@@ -243,6 +358,62 @@ def perform_create(self, serializer):
         user = self.request.user
         serializer.save(creator=user)
 
+    # overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_update(self, request, *args, **kwargs):
+        """
+        Correctly formats bulk PUT/PATCH response
+        """
+        response = super(NodeList, self).bulk_update(request, *args, **kwargs)
+        return Response({'data': response.data}, status=status.HTTP_200_OK)
+
+    # Overrides ListBulkCreateUpdateDestroyAPIView
+    def bulk_destroy(self, request, *args, **kwargs):
+        """
+        Handles bulk destroy of nodes. Handles permissions and enforces bulk limit.
+        """
+        user = self.request.user
+        node_list = []
+        object_type = 'nodes'
+
+        if not request.data:
+            raise ValidationError('Request must contain array of resource identifier objects.')
+
+        if not isinstance(request.data, list):
```

Comment:
A lot of the logic here and in `bulk_update` is duplicated across every subclass of `ListBulk...View`. Would it make sense to create our own base ListBulk...View to DRY up this logic, adding hooks where necessary?